packages/primitives-core/src/styled.ts

```diff
@@ -77,7 +82,8 @@ export function createStyled(
 
         return React.createElement(finalTag, newProps)
       })
-      Styled.withComponent = (newComponent /*: React.ElementType */) =>
+
+      ;(Styled as any).withComponent = (newComponent: React.ElementType) =>
```

Comment:
Because of this cast, the final generated type for this won't see this `withComponent` property. I assume that we need to generate it but maybe I'm wrong.

I would probably go with something like:
```ts
let Styled = // ...

const withComponent = (newComponent: React.ElementType) =>  // ...

const castedStyled = Styled as typeof Styled & { withComponent: typeof withComponent }

castedStyled.withComponent = withComponent

return castedStyled 
```

Reply:
Nice!